### DISCUSSION TITLE
Slow throughput when using as exit node. Hardware offload?

### DISCUSSION THREAD

**Discussion by yoship90** [jamesog/tailscale-edgeos#39] [General]
When using an Edgerouter as an exit node, throughput diminishes significantly (~285mbps without exit node, ~10-35mbps through exit node). Hardware offloading is enabled on the Edgerouter (ipv4/ipv6 forwarding and bonding), but it seems that Tailscale traffic is not offloaded, and therefore tailscale uses a ton of CPU and this seems to be the cause of the very limited throughput.

Does tailscale use PPPoE? Hardware offloading for PPPoE is currently disabled (because VLAN offloading is enabled instead). 

Thanks!

**Comment by jamesog** [comment]
Tailscale is using a userland implementation of Wireguard which is why it's on-CPU. The EdgeRouter platform isn't officially supported so I doubt they'll add hardware offloading, but that's something you should raise with Tailscale directly.

**Comment by yoship90** [comment]
Gotcha, thanks!
